pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -169,25 +173,31 @@ func buildOutboundNetworkFiltersWithWeightedClusters(env *model.Environment, nod
 		}
 	}
 
+	// TODO: Need to handle multiple cluster names for Redis
+	clusterName := clusterSpecifier.WeightedClusters.Clusters[0].Name
 	tcpFilter := setAccessLogAndBuildTCPFilter(env, node, proxyConfig)
-	return buildNetworkFiltersStack(node, port, tcpFilter, statPrefix)
+	return buildNetworkFiltersStack(node, port, tcpFilter, statPrefix, clusterName)
 }
 
 // buildNetworkFiltersStack builds a slice of network filters based on
 // the protocol in use and the given TCP filter instance.
-func buildNetworkFiltersStack(node *model.Proxy, port *model.Port, tcpFilter *listener.Filter, statPrefix string) []listener.Filter {
+func buildNetworkFiltersStack(node *model.Proxy, port *model.Port, tcpFilter *listener.Filter, statPrefix string, clusterName string) []listener.Filter {
 	filterstack := make([]listener.Filter, 0)
 	switch port.Protocol {
 	case model.ProtocolMongo:
-		filterstack = append(filterstack, buildMongoFilter(statPrefix, util.IsProxyVersionGE11(node)), *tcpFilter)
+		filterstack = append(filterstack, buildMongoFilter(statPrefix, util.IsXDSMarshalingToAnyEnabled(node)))
+	case model.ProtocolRedis:
+		if util.IsProxyVersionGE11(node) && pilot.EnableRedisFilter() {
+			filterstack = append(filterstack, buildRedisFilter(statPrefix, clusterName, util.IsXDSMarshalingToAnyEnabled(node)))
+		}
 	case model.ProtocolMySQL:
 		if util.IsProxyVersionGE11(node) && pilot.EnableMysqlFilter() {
-			filterstack = append(filterstack, buildMySQLFilter(statPrefix, util.IsProxyVersionGE11(node)))
+			filterstack = append(filterstack, buildMySQLFilter(statPrefix, util.IsXDSMarshalingToAnyEnabled(node)))
 		}
-		filterstack = append(filterstack, *tcpFilter)
-	default:
-		filterstack = append(filterstack, *tcpFilter)
 	}
+
+	filterstack = append(filterstack, *tcpFilter)
+
```

Comment:
We can revert the `filterstack = append(filterstack, *tcpFilter)` change here, as the Redis filter should ideally be the last one in the filter stack.

Reply:
Isn't this same as MYSQL and MONGO?

Reply:
We talked in private, redis proxy will do some route, while mysql and mongo do not.